python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -72,6 +74,7 @@ def asset(
     description: Optional[str] = None,
     required_resource_keys: Optional[Set[str]] = None,
     resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
+    io_manager: Optional[Union[Mapping[str, IOManagerDefinition], str, IOManagerDefinition]] = None,
```

Comment:
I think this should be called io_manager_def, to match resource_defs etc.

Reply:
If we call this io_manager_def, I think we should keep around io_manager_key and have people specify one or the other.